### PR TITLE
feat: Create module documentation

### PR DESCRIPTION
**Pull request by Darkheir**
First documentation about the creation of modules.

It contains 2 new pages:
* An overview of the playbook modules explaining the main concepts: [page](https://storage.gra.cloud.ovh.net/v1/AUTH_741841e1c7da4c50a76152cbc2609bd1/sekoiaio-documentation-previews/932/xdr/develop/guides/playbook_modules/overview/)
* A quick start explaining how to create a new module: [page](https://storage.gra.cloud.ovh.net/v1/AUTH_741841e1c7da4c50a76152cbc2609bd1/sekoiaio-documentation-previews/932/xdr/develop/guides/playbook_modules/quick_start/)

### CODE REVIEW

File: _shared_content/develop/guides/playbook_modules/overview.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
+# Overview
+
+## What is a playbook module
```
Comment (otetard) [inline _shared_content/develop/guides/playbook_modules/overview.md:3]:
```suggestion
## What is a playbook module?
```

File: _shared_content/develop/guides/playbook_modules/overview.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+* A manifest that describes the module and the expected configuration
+* For each action and trigger a manifest file describing the expected arguments and the results 
+* The code to execute
+
+## The manifest file
+
+Each module contains a `manifest.json` file that describes the module. This file contains:
+
+* The name of the module
+* A unique uuid identifying the module
```
Comment (otetard) [inline _shared_content/develop/guides/playbook_modules/overview.md:18]:
```suggestion
* A unique UUID identifying the module
```

File: _shared_content/develop/guides/playbook_modules/overview.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+* A version
+* A description
+* The configuration of the module
+    * All the informations that are needed for the module to be able to communicate with the external service
+    * The `configuration` key should be a valid json schema object
```
Comment (otetard) [inline _shared_content/develop/guides/playbook_modules/overview.md:23]:
```suggestion
    * The `configuration` key should be a valid JSON schema object
```

File: _shared_content/develop/guides/playbook_modules/overview.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+Here's an example of a manifest:
```
Comment (otetard) [inline _shared_content/develop/guides/playbook_modules/overview.md:25]:
```suggestion
Here's an example of a `manifest.json` file:
```

File: _shared_content/develop/guides/playbook_modules/overview.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+Each trigger/action has its own manifest file that describes it. This manifest file is a json file prefixed by `trigger_`  or `action_` followed by the slug of the trigger/action.
+
+i.e. The manifest for the action allowing to download a file is named `action_download_file.json`
+
+The manifest contains:
+
+* The name of the trigger/action
+* A unique uuid
+* A description
+* The docker parameters that will be specified when running the trigger/action
```
Comment (otetard) [inline _shared_content/develop/guides/playbook_modules/overview.md:91]:
```suggestion
* The Docker parameters that will be specified when running the trigger/action
```

File: _shared_content/develop/guides/playbook_modules/overview.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+### Manifest
+
+Each trigger/action has its own manifest file that describes it. This manifest file is a json file prefixed by `trigger_`  or `action_` followed by the slug of the trigger/action.
+
+i.e. The manifest for the action allowing to download a file is named `action_download_file.json`
+
+The manifest contains:
+
+* The name of the trigger/action
+* A unique uuid
```
Comment (otetard) [inline _shared_content/develop/guides/playbook_modules/overview.md:89]:
```suggestion
* A unique UUID
```

File: _shared_content/develop/guides/playbook_modules/overview.md [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+* A `pyproject.toml` and a `poetry.lock` file that allow to define all the requirements for the module 
+    * You can check the [Poetry documentation](https://python-poetry.org/ ) for more information
+
+#### Module entrypoint
+
+The `main.py` file is the only entrypoint in the module actions and trigger. 
+It will execute the appropriate action or trigger depending on the argument that is provided to the command.
+This argument is defined by the `docker_parameters` attribute in the manifest files.
+
+The sekoia automation SDK makes it easier to declare the trigger and actions that can be executed.
```
Comment (otetard) [inline _shared_content/develop/guides/playbook_modules/overview.md:167]:
```suggestion
The `sekoia-automation-sdk` PIP package makes it easier to declare the trigger and actions that can be executed.
```

File: _shared_content/develop/guides/playbook_modules/overview.md [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+    module.run()
+```
+
+## Additional files
+
+### Logo
+
+Additionally a module should contain a logo image file. The file should be name either `logo.png` or `logo.svg` and located at the root of the module
+
+### Dockerfile
```
Comment (otetard) [inline _shared_content/develop/guides/playbook_modules/overview.md:190]:
```suggestion
### `Dockerfile`
```

File: _shared_content/develop/guides/playbook_modules/overview.md [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+
+The dockerfile allows to create an image that will contain all the required code and dependencies to run the module's actions and triggers.
```
Comment (otetard) [inline _shared_content/develop/guides/playbook_modules/overview.md:192]:
```suggestion
The `Dockerfile` allows to create an image that will contain all the required code and dependencies to run the module's actions and triggers.
```

File: _shared_content/develop/guides/playbook_modules/quick_start.md [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+poetry lock
+```
+
+!!! note
+    If poetry is not installed on your system you may need to run `pip install poetry` before.
+
+
+#### Create the action
+
+Inside our python package `testhttp_modules` we can create a new file `action_request.py` that will contain the logic of our action:
```
Comment (otetard) [inline _shared_content/develop/guides/playbook_modules/quick_start.md:164]:
```suggestion
Inside our Python package `testhttp_modules` we can create a new file `action_request.py` that will contain the logic of our action:
```

**Review by otetard — APPROVED** [review]
Great! I just made some typo remarks (#TypoNazi)

**Comment by otetard** [comment]
You can also use MkDocs Material features to annotate code sections: https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#code-annotations

**Comment by Darkheir** [comment]
> You can also use MkDocs Material features to annotate code sections: https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#code-annotations

Awesome, I will update the pages using this feature!

File: _shared_content/develop/guides/playbook_modules/overview.md [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+  }
+}
+```
+
+## Triggers and Actions
+
+### Trigger
+
+A trigger is some code that runs indefinitely waiting for events to occur. Once an event happens the trigger will push it to the playbook API.
+Those events will trigger new playbook runs for all the palybooks that have the trigger as an entrypoint.
```
Comment (gaelmuller) [inline _shared_content/develop/guides/playbook_modules/overview.md:62]: Palybooks

File: _shared_content/develop/guides/playbook_modules/overview.md [hunk 12]
```diff
@@ -0,0 +1,213 @@
...
+    module.register(MWDBConfigsTrigger, "trigger_mwdb_configs")
+    module.register(ConfigToObservablesAction, "config_to_observables")
+    module.run()
+```
+
+## Additional files
+
+### Logo
+
+Additionally a module should contain a logo image file. The file should be name either `logo.png` or `logo.svg` and located at the root of the module
```
Comment (gaelmuller) [inline _shared_content/develop/guides/playbook_modules/overview.md:188]: Named

File: _shared_content/develop/guides/playbook_modules/quick_start.md [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+Simply run the following command:
+
+```shell
+pip install sekoia-automation-sdk
+```
+
+!!! note
+    The SDK needs a python version equal or newer to python3.10
+
+Once the SDK is installed we can use it to crea an empty module:
```
Comment (gaelmuller) [inline _shared_content/develop/guides/playbook_modules/quick_start.md:24]: Create

File: _shared_content/develop/guides/playbook_modules/quick_start.md [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+│   └── models.py
+└── tests
+    ├── conftest.py
+    └── __init__.py
+```
+
+
+## Create our action
+
+Out action will take three arguments:
```
Comment (gaelmuller) [inline _shared_content/develop/guides/playbook_modules/quick_start.md:54]: Our

File: _shared_content/develop/guides/playbook_modules/quick_start.md [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+
+* The URL to query
+* The HTTP method to use
+* The HTTP headers to provide
+
+Two of those arguemtns will be required, the URL and the method.
```
Comment (gaelmuller) [inline _shared_content/develop/guides/playbook_modules/quick_start.md:60]: Arguments

File: _shared_content/develop/guides/playbook_modules/quick_start.md [hunk 16]
```diff
@@ -0,0 +1,234 @@
...
+
+The result of our action will contain:
+
+* The status code
+* The response body as text
+* The response headers
+
+### Define the manifest
+
+Let's create an `action_request.json` file that describe out action:
```
Comment (gaelmuller) [inline _shared_content/develop/guides/playbook_modules/quick_start.md:70]: Our

File: _shared_content/develop/guides/playbook_modules/quick_start.md [hunk 17]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+```
+
+### Add the code
+
+Now that we have a manifest describing our action we can add the code!
+
+
+#### Specify the requirements
```
Comment (gaelmuller) [inline _shared_content/develop/guides/playbook_modules/quick_start.md:143]: Why not use `poetry add requests` here ?

File: _shared_content/develop/guides/playbook_modules/quick_start.md [hunk 18]
```diff
@@ -0,0 +1,234 @@
...
+        return {
+            "status_code": response.status_code,
+            "headers": dict(response.headers),
+            "text": response.text,
+        }
+```
+
+Let's explain a bit the code of the action.
+
+Our action is a class that inherit from `Action`, a base class from SEKOIA's automation SDK.
```
Comment (gaelmuller) [inline _shared_content/develop/guides/playbook_modules/quick_start.md:202]: SEKOIA.IO

File: _shared_content/develop/guides/playbook_modules/quick_start.md [hunk 19]
```diff
@@ -0,0 +1,234 @@
...
+    module = Module()
+
+    module.register(RequestAction, "request")
+
+    module.run()
+```
+
+The action has been registered between the creation of the module and the call running it. The first arguments of `module.register` is our action class and the second is the `docker_parameter` that was specified in our action's manifest.
+
+That it! Now we have a module containing our action and ready to run !
```
Comment (gaelmuller) [inline _shared_content/develop/guides/playbook_modules/quick_start.md:234]: That's it

File: _shared_content/develop/guides/playbook_modules/quick_start.md [hunk 20]
```diff
@@ -0,0 +1,234 @@
+# Quick start
+
+In this section we will see how to create a simple module from scratch.
+
+This module will conain one action and one trigger:
+
+* The action will allow to send an HTTP request and return its response.
+* The trigger will watch for ??
```
Comment (gaelmuller) [inline _shared_content/develop/guides/playbook_modules/quick_start.md:8]: Fixme

**Review by gaelmuller — CHANGES_REQUESTED** [review]
Nice 👍

Potential improvements:

* Use pydantic model and auto-generate manifestes
* Use the module configuration inside the action

Requested changes are for the typos

File: _shared_content/develop/guides/playbook_modules/overview.md (hunk 12 above)
Comment (Darkheir) [reply _shared_content/develop/guides/playbook_modules/overview.md:188]:
```suggestion
Additionally a module should contain a logo image file. The file should be named either `logo.png` or `logo.svg` and located at the root of the module
```

File: _shared_content/develop/guides/playbook_modules/quick_start.md (hunk 13 above)
Comment (Darkheir) [reply _shared_content/develop/guides/playbook_modules/quick_start.md:24]:
```suggestion
Once the SDK is installed we can use it to create an empty module:
```

File: _shared_content/develop/guides/playbook_modules/quick_start.md (hunk 16 above)
Comment (Darkheir) [reply _shared_content/develop/guides/playbook_modules/quick_start.md:70]:
```suggestion
Let's create an `action_request.json` file that describe our action:
```

File: _shared_content/develop/guides/playbook_modules/quick_start.md (hunk 18 above)
Comment (Darkheir) [reply _shared_content/develop/guides/playbook_modules/quick_start.md:202]:
```suggestion
Our action is a class that inherit from `Action`, a base class from SEKOIA.IO's automation SDK.
```

File: _shared_content/develop/guides/playbook_modules/quick_start.md (hunk 20 above)
Comment (Darkheir) [reply _shared_content/develop/guides/playbook_modules/quick_start.md:8]: I'm looking for an idea of q simple trigger we could add to the quick start :sweat_smile:

**Comment by Darkheir** [comment]
I just updated the pages to use code annotation, use Pydantic schemas and generate the action manifest and the entrypoint.

**Comment by gbossert** [comment]
I've added few suggestions to the PR

File: _shared_content/develop/guides/automation/create_a_module.md [hunk 21]
```diff
@@ -0,0 +1,403 @@
...
+This SDK contains various command line utilities that will help us manage modules.
+
+Simply run the following command:
+
+```shell
+pip install sekoia-automation-sdk
+```
+
+!!! note
+    The SDK needs a python version equal or newer to python3.10
```
Comment (otetard) [inline _shared_content/develop/guides/automation/create_a_module.md:23]:
```suggestion
    The SDK needs a Python version equal or newer to 3.10.
```

File: _shared_content/develop/guides/automation/overview.md [hunk 22]
```diff
@@ -0,0 +1,223 @@
...
+└── tests/  // (10)!
+```
+
+1. the module manifest file that specifies elements such as the name or the module's configuration
+2. the manifest file of the action `enrich_ip`
+3. the manifest file of the trigger `trigger_pull_logs`
+4. the logo of the module
+5. the code entrypoint
+6. the poetry dependency listing file
+7. the python project definition
```
Comment (otetard) [inline _shared_content/develop/guides/automation/overview.md:42]:
```suggestion
7. the Python project definition
```

File: _shared_content/develop/guides/automation/overview.md [hunk 23]
```diff
@@ -0,0 +1,223 @@
...
+├── cyber_enricher/  // (9)!
+└── tests/  // (10)!
+```
+
+1. the module manifest file that specifies elements such as the name or the module's configuration
+2. the manifest file of the action `enrich_ip`
+3. the manifest file of the trigger `trigger_pull_logs`
+4. the logo of the module
+5. the code entrypoint
+6. the poetry dependency listing file
```
Comment (otetard) [inline _shared_content/develop/guides/automation/overview.md:41]:
```suggestion
6. the Poetry dependency listing file
```

**Pull request merged**